pkg/operator/ceph/cluster/predicate.go

```diff
@@ -149,6 +149,12 @@ func watchControllerPredicate(ctx context.Context, c client.Client) predicate.Fu
 						return false
 					}
 
+					err := objNew.Spec.Storage.ValidateOSDStoreUpdate(objOld.Spec.Storage)
+					if err != nil {
+						logger.Warningf("skipping orchestration. %v", err)
```

Comment:
Why is it skipped here? Because they are trying to convert invalid OSDs? What if instead we just ignore the conversion if it's invalid. There simply would be no OSDs in need of conversion, right?

Reply:
This check was to do some validation on the `osdstore` related updates that user have made in the cephcluster. For example, not allowing the change from `bluestore-rdr` to `bluestore`. 
But we need a better way to implement this change. So I'v removed these changes for now.